modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -0,0 +1,162 @@
+        keySofar.current = '';
+      }, 500);
+    };
+
+    // Based on the the key typed, this will try and find an index in the array where it matches the string
+    // so if a user types `de` it will try and find an index in the array whose id matches `de` like denver
+    const getIndexByStartString = (
+      startIndex: number,
+      startString: string,
+      endIndex: number = model.state.items.length,
+      ignoreDisabled: boolean = true
+    ): number => {
+      for (let i = startIndex; i < endIndex; i++) {
+        const label = model.state.items[i].id.toLowerCase();
+
+        if (label.indexOf(startString.toLowerCase()) === 0) {
+          if (
+            !ignoreDisabled ||
+            (ignoreDisabled && !model.state.nonInteractiveIds.includes(model.state.items[i].id))
+          ) {
+            return i;
+          }
+        }
+      }
+      return -1;
+    };
+    // just use the index you dummy squid
+    const currentItem = model.navigation.getItem(model.state.cursorId, model);
+    const cursorFocusedIndex = model.state.items.findIndex(item => item.id === currentItem.id);
+
+    const handleKeyboardTypeAhead = (key: string, numOptions: number) => {
+      // If the starting point is beyond the list of options, reset it
+      // to the beginning of the list
+      let start = keySofar.current.length === 0 ? cursorFocusedIndex + 1 : cursorFocusedIndex;
+
+      start = start === numOptions ? 0 : start;
+
+      // Keeps track of the current key types and adds to it
+      // if you type `de` vs `d` for denver
+      keySofar.current += key;
+      startClearKeysSoFarTimer();
+
+      // First, look for a match from start to end
+      let matchIndex;
+      matchIndex = getIndexByStartString(start, keySofar.current);
+
+      // If a match isn't found between start and end, wrap the search
+      // around and search again from the beginning (0) to start
+      if (matchIndex === -1) {
+        matchIndex = getIndexByStartString(0, keySofar.current, start);
+      }
+
+      // A match was found...
+      if (matchIndex > -1) {
+        if (model.state.visibility === 'hidden') {
+          // If the menu is closed, fire the change event
+          // go to that item and select based on its id
+          model.events.goTo({id: model.state.items[matchIndex].id});
+          model.events.select({id: model.state.items[matchIndex].id});
+        } else {
+          // Otherwise the menu is visible
+          // move the cursor to that matched item
+          model.events.goTo({id: model.state.items[matchIndex].id});
+        }
+      }
+    };
+
+    return {
+      onKeyDown(event: React.KeyboardEvent) {
+        const foundIndex = model.state.items.findIndex((item: {id: string}) => {
+          if (model.state.selectedIds.length > 0) {
+            return item.id === model.state.selectedIds[0];
+          } else {
+            return item.id === model.state.cursorId;
+          }
+        });
```

Comment:
```suggestion
        const foundIndex = model.state.items.findIndex(
          (item: {id: string}) => 
            (model.state.selectedIds.length > 0 && item.id === model.state.selectedIds[0]) || item.id === model.state.cursorId
        );
```
another way to set condition